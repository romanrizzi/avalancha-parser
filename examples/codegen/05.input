
fun sucsuc
  x -> Suc(Suc(x))

fun if
  True,  x, _ -> x
  False, _, y -> y

print sucsuc(sucsuc(Zero))
print if(True, A, B)
print if(False, A, B)

