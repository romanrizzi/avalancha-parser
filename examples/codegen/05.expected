Suc(Suc(Suc(Suc(Zero))))
A
B
